models/src/anemoi/models/compile/decorators.py

```diff
@@ -0,0 +1,22 @@
+# (C) Copyright 2024 Anemoi contributors.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+def torch_compile():
+    def decorator(func):
+        import torch
+        import torch_geometric
+
+        req = torch.__version__ >= "2.6" and torch_geometric.__version__ >= "2.6"
```

Comment:
Agreed that we should make the minimal version of torch be `2.6` and remove this.